File: test/lit/zig/if_expr.zig
// RUN: %cot emit-cir %s | %FileCheck %s

pub fn clamp_low(x: i32) i32 {
    const result = if (x > 0) x else 0;
    return result;
}

// CHECK-LABEL: func.func @clamp_low
// CHECK: cir.cmp sgt
// CHECK: cir.select

File: test/lit/ts/if_expr.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Feature #016: If expression (ternary) — TypeScript syntax

function max(a: number, b: number): number {
    return a > b ? a : b;
}

// CHECK-LABEL: func.func @max
// CHECK: cir.cmp sgt
// CHECK: cir.select
// CHECK: return
